### PR TITLE
ZK-4564: split zkau servlet into dedicated au and resource servlets

### PR DESCRIPTION
**Pull request by DevChu**

### CODE REVIEW

File: zk/src/org/zkoss/zk/ui/sys/UiFactory.java [hunk 1]
```diff
@@ -96,10 +96,11 @@
 	 * @param updateURI the URI that is used as the base to generate
 	 * URI for asynchronous updates. In other words, it is used to
 	 * implement {@link Desktop#getUpdateURI}
+	 * @param resourceURI the URI that is used to get ZK resource
 	 * @param path the path to request this desktop, such as "/test/test.zul",
 	 * or null if it is caused by a filter.
 	 */
-	public Desktop newDesktop(RequestInfo ri, String updateURI, String path);
+	public Desktop newDesktop(RequestInfo ri, String updateURI, String resourceURI, String path);
```
Comment (DevChu) [inline zk/src/org/zkoss/zk/ui/sys/UiFactory.java:103]: overload

File: zktest/src/archive/test2/F92-ZK-4564.zul [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+-->
+<zk>
+	<label>click the button and the operate with the following components, there should be no js error/exceptions</label>
+	<button label="click" onClick='Clients.log("Clicked!")' />
+	<fileupload />
+	<pdfviewer />
+	<tbeditor />
+	<imagemap id="img1" src="/test2/img/sun.jpg"/>
+	<script>
+		jq.getScript('/zktest/zkau/web/js/zk.wpd')
```
Comment (DevChu) [inline zktest/src/archive/test2/F92-ZK-4564.zul:21]: use zk.ajaxURI('...', {au: true})

**Comment by scribetw** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**